### PR TITLE
Interactivity guide

### PR DESCRIPTION
**Pull request by CrystalOnScript**
'bout time! More examples to come in the future.

### CODE REVIEW

File: pages/content/amp-dev/documentation/guides-and-tutorials/develop/animations/_blueprint.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ $path: /documentation/guides-and-tutorials/develop/animations/{base}.html
 $localization:
   path: /{locale}/documentation/guides-and-tutorials/develop/animations/{base}.html
 $title@: Animate & Transition
-$order: 1
+$order: 3
```
Comment (caroqliu) [inline pages/content/amp-dev/documentation/guides-and-tutorials/develop/animations/_blueprint.yaml:6]: FMI: Where does this ordering matter?

File: pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/foundations.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+ <h1 id="togleHeading">Hello AMP World!</h1>
+  <button on="tap:togleHeading.toggleClass(class='red'),togleHeading.toggleClass(class='green', force=false)">
+    Toggle Red
+  </button>
+  <button on="tap:togleHeading.toggleClass(class='green'),togleHeading.toggleClass(class='red', force=false)">
+    Toggle Green
+  </button>
+  <button on="tap:togleHeading.toggleClass(class='border')">
+    Toggle Border
+  </button>
```
Comment (caroqliu) [inline pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/foundations.md:191]:
```suggestion
<button on="tap:togleHeading.toggleClass(class='red'),togleHeading.toggleClass(class='green', force=false)">
  Toggle Red
</button>
<button on="tap:togleHeading.toggleClass(class='green'),togleHeading.toggleClass(class='red', force=false)">
  Toggle Green
</button>
<button on="tap:togleHeading.toggleClass(class='border')">
  Toggle Border
</button>
```

File: pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/index.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+$title: Interactivity in AMP introduction
+$order: 0
+description: "Read this guide to learn how to build interactivity in AMP."
+$hidden: true
+author: CrystalOnScript
+---
+
+These guides outline how to create interactive AMP pages. Interactivity is essential to the modern web. Building unique and delightful experiences engages users, keeping them on your site. AMP provides out-of-the box solutions to common interactivity needs while also affording highly customizable and flexible solutions.
+
+Building interactive experiences in AMP may feel untraditional, and it is! But that doesn’t mean it’s more difficult. Many interactive experiences are very easy to build with AMP once you’re introduced to concepts and understand the syntax.
```
Comment (caroqliu) [inline pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/index.md:13]: Optional: Something about the "It is <not traditional>" is a bit confusing for me, can we instead rephrase to say "It is not <traditional>", i.e.  `"Building interactive experiences may not feel traditional, and it's not"`

File: pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/index.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+```
+[/example]
+
+Use [this guide](custom_experiences.md) as an introduction to the differences between the three, learn if and how to combine them with each other and use with page elements.
+
+# Client-side rendering
+
+This guide outlines client-side rendering possibilities in AMP. By default, AMP pages are server-side rendered. However, in certain cases it's necessary to dynamically render data on the client, such as a list of articles or items in a user’s shopping cart.
+
+Example (?)
```
Comment (caroqliu) [inline pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/index.md:102]: Should this be replaced?

File: pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/index.md [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  <button on="tap:sidebar.toggle">Toggle sidebar</button>
+  <button on="tap:sidebar.open">Open sidebar</button>
+```
+[/example]
+
+[This guide](ready_made.md) introduces the most common interactive components and outlines implementation patterns.
+
+# Building custom interactive experiences
+
+While AMP provides solutions to common web widgets and interactivity, each website has its own needs. AMP embraces highly customizable experiences with three components: [`amp-selector`](../../../components/reference/amp-selector.md), [`amp-bind`](../../../components/reference/amp-bind.md), and [`amp-script`](../../../components/reference/amp-bind.md).
```
Comment (caroqliu) [inline pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/index.md:66]: To me `amp-selector` is not obviously in the same category as `amp-bind` and `amp-script`, maybe elaborate more on why these make sense to group together?

File: pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/ready_made.md [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+  - sbenz
+---
+
+This guide outlines interactive possibilities available in the  [AMP component library](../../../components/index.html). The library provides ready-made elements that meet common and uncommon user interface needs. You are able to customize these components, and how they interact with users, to meet specific needs. Read-on to learn how to harness this to decrease initial build costs without losing adaptability.
+
+# Ready-made interactive components
+
+Many of the available components fulfill a single purpose of interactivity. Most can standalone, but a core principle of AMP is composability. Composability allows you to combine different AMP components together to implement a customized interaction. Most allow templating with [amp-mustache](https://amp.dev/documentation/components/amp-mustache/).
+
+AMP offers the following ready-made interactive components:
```
Comment (caroqliu) [inline pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/ready_made.md:18]: Suggestion: We could also include `amp-app-banner`, `amp-image-lightbox`, `amp-image-slider`, `amp-truncate-text`, `amp-video`, `amp-video-docking`

**Review by caroqliu — APPROVED** [review]
Added some comments and suggestions, but overall these contributions are so comprehensive, clear, and helpful. Thank you!!

File: pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/custom_experiences.md [hunk 7]
```diff
@@ -0,0 +1,280 @@
...
+  - websites
+$title: Building custom interactive experiences
+$order: 3
+description: "This guide outlines highly customizable interactivity options to integrate into AMP pages."
+author: CrystalOnScript
+contributors:
+  - sbenz
+---
+
+This guide outlines highly customizable interactivity options to integrate into AMP pages.  Like snowflakes and variable names, each website is unique and AMP's ready-made component library may not meet all your needs. Instead, AMP provides adaptable components that enable personalized solutions to complex interactivity needs. Use these components to build selection interfaces, widget, and the ability to process custom logic.
```
[the remark this answers is not in the document]
Comment (CrystalOnScript) [reply pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/custom_experiences.md:12]: Yes, great catch - thanks!

File: pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/custom_experiences.md [hunk 8]
```diff
@@ -0,0 +1,280 @@
...
+</amp-state>
+<p [text]="foo.bar">No state is set</p>
+<button on="tap:AMP.setState({})">Set State</button>
+
+```
+[/example]
+
+The `amp-bind` component performs no evaluations on page load, with the exception of combining with amp-list.
+
+Read on: Read more on evaluating state on page load with amp-list in Client-side rendering in AMP.
```
[the remark this answers is not in the document]
Comment (CrystalOnScript) [reply pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/custom_experiences.md:116]: thanks for the catch - fixed the UI and added a link

File: pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/foundations.md [hunk 9]
```diff
@@ -0,0 +1,196 @@
...
+  <button on="tap:helloWorld.hide">
+    Goodbye World
+  </button>
+```
+[/example]
+
+Event listeners and actions follow a specific syntax in AMP:
+
+```html
+on="eventName:targetId.actionName"
```
[the remark this answers is not in the document]
Comment (CrystalOnScript) [reply pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/foundations.md:30]: That's even better - thank you!

File: pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/index.md (hunk 3 above)
Comment (CrystalOnScript) [reply pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/index.md:13]: I agree - the rewrite is much better. Thanks!

File: pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/index.md (hunk 5 above)
Comment (CrystalOnScript) [reply pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/index.md:66]: added a bit more information, hopefully that helps!

File: pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/ready_made.md (hunk 6 above)
Comment (CrystalOnScript) [reply pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/ready_made.md:18]: Fantastic, thanks for flagging these!

File: pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/custom_experiences.md [hunk 10]
```diff
@@ -0,0 +1,284 @@
+---
+formats:
+  - websites
+$title: Building custom interactive experiences
```
Comment (nainar) [inline pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/custom_experiences.md:4]: can we name the file personalized experiences - that will SEO better

Comment (CrystalOnScript) [reply pages/content/amp-dev/documentation/guides-and-tutorials/develop/interactivity_guide/custom_experiences.md:4]: you got it ;)

**Pull request merged**